firebase-storage/src/test/java/com/google/firebase/storage/UploadTest.java

```diff
@@ -137,33 +129,25 @@ public void cantUploadToRoot() throws Exception {
           taskException.set(exception);
         });
 
-    for (int i = 0; i < 300000; i++) {
-      Robolectric.flushForegroundThreadScheduler();
-      if (taskException.get() != null) {
-        // Test the various exception accessors.
-        try {
-          task.getResult();
-          Assert.fail();
-        } catch (RuntimeExecutionException e) {
-          Assert.assertEquals(taskException.get().getCause(), e.getCause().getCause());
-          // Task failed.
-        }
-
-        try {
-          task.getResult(StorageException.class);
-          Assert.fail();
-        } catch (StorageException e) {
-          Assert.assertEquals(taskException.get().getCause(), e.getCause());
-          // Task failed.
-        }
-
-        Assert.assertEquals(taskException.get().getCause(), task.getException().getCause());
-
-        return;
-      }
-      Thread.sleep(1);
+    TestUtil.await(task, 300, TimeUnit.SECONDS);
```

Comment:
Note: Some of the timeouts are VERY questionable, but I kept them as is to make this purely a refactor.

Reply:
Yeah, this looks a little insane. However, this test looks to be exercising network failures, so, as an integration test, it might not be unreasonable for this to take multiple minutes to fail. But it's a unit test; so hopefully it doesn't actually take anywhere near that long. (Maybe 5m was used when writing the test against the real network? Then it was converted to a unit test, but the timeout wasn't fixed?)

Regardless, keeping it the same to keep this PR as a refactor sounds reasonable. You could optionally add a TODO to investigate whether this value still makes sense. (If you do, write the todo to apply to all non default timeouts.)

Reply:
I added TODOs to all timeouts greater than 5 seconds with the exception of those that were already commented.